### PR TITLE
K8SPG-138 fix repo host

### PR DESCRIPTION
**Pull request by nmarukovich**
[K8SPG-138](https://jira.percona.com/browse/K8SPG-138) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
*Short explanation of the problem.*

**Cause:**
*Short explanation of the root cause of the issue if applicable.*

**Solution:**
*Short explanation of the solution we are providing with this PR.*

**CHECKLIST**
---
**Jira**
- [ ] Is the Jira ticket created and referenced properly?
- [ ] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [ ] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PG version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

[K8SPG-138]: https://perconadev.atlassian.net/browse/K8SPG-138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/custom-extensions.log) |
| demand-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/demand-backup.log) |
| init-deploy | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/init-deploy.log) |
| monitoring | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/monitoring.log) |
| operator-self-healing | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/operator-self-healing.log) |
| scaling | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/scaling.log) |
| scheduled-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/scheduled-backup.log) |
| self-healing | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/self-healing.log) |
| start-from-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/start-from-backup.log) |
| telemetry-transfer | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/telemetry-transfer.log) |
| upgrade-minor | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/upgrade-minor.log) |
| users | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0e82894fb/users.log) |
| We run 4 out of 12|

commit: https://github.com/percona/percona-postgresql-operator/pull/727/commits/0e82894fb3a3c8b3e711fe069919a4eb04e09a9b
image: `perconalab/percona-postgresql-operator:PR-727-0e82894fb`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/custom-extensions.log) |
| demand-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/demand-backup.log) |
| init-deploy | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/init-deploy.log) |
| monitoring | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/monitoring.log) |
| operator-self-healing | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/operator-self-healing.log) |
| scaling | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/scaling.log) |
| scheduled-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/scheduled-backup.log) |
| self-healing | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/self-healing.log) |
| start-from-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/start-from-backup.log) |
| telemetry-transfer | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/telemetry-transfer.log) |
| upgrade-minor | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/upgrade-minor.log) |
| users | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/58b3c20de/users.log) |
| We run 4 out of 12|

commit: https://github.com/percona/percona-postgresql-operator/pull/727/commits/58b3c20deff09c5680ac1be8bd4ae75066641a7b
image: `perconalab/percona-postgresql-operator:PR-727-58b3c20de`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/custom-extensions.log) |
| demand-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/demand-backup.log) |
| init-deploy | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/init-deploy.log) |
| monitoring | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/monitoring.log) |
| operator-self-healing | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/operator-self-healing.log) |
| scaling | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/scaling.log) |
| scheduled-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/scheduled-backup.log) |
| self-healing | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/self-healing.log) |
| start-from-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/start-from-backup.log) |
| telemetry-transfer | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/telemetry-transfer.log) |
| upgrade-minor | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/upgrade-minor.log) |
| users | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/5a57a8dbb/users.log) |
| We run 4 out of 12|

commit: https://github.com/percona/percona-postgresql-operator/pull/727/commits/5a57a8dbb71410238f20b9423e11a48436dc9b79
image: `perconalab/percona-postgresql-operator:PR-727-5a57a8dbb`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/custom-extensions.log) |
| demand-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/demand-backup.log) |
| init-deploy | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/init-deploy.log) |
| monitoring | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/monitoring.log) |
| operator-self-healing | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/operator-self-healing.log) |
| scaling | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/scaling.log) |
| scheduled-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/scheduled-backup.log) |
| self-healing | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/self-healing.log) |
| start-from-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/start-from-backup.log) |
| telemetry-transfer | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/telemetry-transfer.log) |
| upgrade-minor | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/upgrade-minor.log) |
| users | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/78353e70f/users.log) |
| We run 4 out of 12|

commit: https://github.com/percona/percona-postgresql-operator/pull/727/commits/78353e70f351280cb71e87644cbe98c60f563ca5
image: `perconalab/percona-postgresql-operator:PR-727-78353e70f`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/custom-extensions.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/scheduled-backup.log) |
| self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/telemetry-transfer.log) |
| upgrade-minor | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/upgrade-minor.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-postgresql-operator/pull/727/commits/a8363933d27d44a4c2a25eb38a6e134847e979ff
image: `perconalab/percona-postgresql-operator:PR-727-a8363933d`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/custom-extensions.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/telemetry-transfer.log) |
| upgrade-minor | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/upgrade-minor.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/a8363933d/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-postgresql-operator/pull/727/commits/a8363933d27d44a4c2a25eb38a6e134847e979ff
image: `perconalab/percona-postgresql-operator:PR-727-a8363933d`

**Review by egegunes — APPROVED** [review]

File: internal/controller/postgrescluster/pgbackrest.go [hunk 1]
```diff
@@ -588,7 +588,10 @@ func (r *Reconciler) generateRepoHostIntent(postgresCluster *v1beta1.PostgresClu
 	// - https://cloud.google.com/kubernetes-engine/docs/concepts/workload-identity
 	// - https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
 	repo.Spec.Template.Spec.AutomountServiceAccountToken = initialize.Bool(false)
-	repo.Spec.Template.Spec.ServiceAccountName = naming.PGBackRestRBAC(postgresCluster).Name
+	currVersion := postgresCluster.Labels[naming.LabelVersion]
+	if currVersion >= "2.4.0" {
```
Comment (egegunes) [inline internal/controller/postgrescluster/pgbackrest.go:592]: does `>=` work with version, better to use our `Version` package

**Review by egegunes — CHANGES_REQUESTED** [review]

**Review by egegunes — DISMISSED** [review]

**Review by egegunes — CHANGES_REQUESTED** [review]
please fix golangci-lint error (run `goimports` on `internal/controller/postgrescluster/pgbackrest.go`)

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/custom-extensions.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/scheduled-backup.log) |
| self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/telemetry-transfer.log) |
| upgrade-minor | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/upgrade-minor.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/91dd17710/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-postgresql-operator/pull/727/commits/91dd177103519e9dd58d1d713d4db4434b7ede90
image: `perconalab/percona-postgresql-operator:PR-727-91dd17710`

**Review by egegunes — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/custom-extensions.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/telemetry-transfer.log) |
| upgrade-minor | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/upgrade-minor.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-727/0b54c411b/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-postgresql-operator/pull/727/commits/0b54c411bb60e32af415b263610ccd892e10598f
image: `perconalab/percona-postgresql-operator:PR-727-0b54c411b`

**Review by hors — APPROVED** [review]

**Review by inelpandzic — APPROVED** [review]

**Pull request merged**